src/main/java/org/spongepowered/api/event/cause/entity/teleport/MovementTypes.java

```diff
@@ -28,24 +28,24 @@
 
 import java.util.function.Supplier;
 
-public final class TeleportTypes {
+public final class MovementTypes {
 
     // SORTFIELDS:ON
 
-    public static final Supplier<TeleportType> COMMAND = Sponge.getRegistry().getCatalogRegistry().provideSupplier(TeleportType.class, "COMMAND");
+    public static final Supplier<MovementType> COMMAND = Sponge.getRegistry().getCatalogRegistry().provideSupplier(MovementType.class, "COMMAND");
 
-    public static final Supplier<TeleportType> ENTITY_TELEPORT = Sponge.getRegistry().getCatalogRegistry().provideSupplier(TeleportType.class, "ENTITY_TELEPORT");
+    public static final Supplier<MovementType> ENDER_PEARL = Sponge.getRegistry().getCatalogRegistry().provideSupplier(MovementType.class, "ENDER_PEARL");
 
-    public static final Supplier<TeleportType> PLUGIN = Sponge.getRegistry().getCatalogRegistry().provideSupplier(TeleportType.class, "PLUGIN");
+    public static final Supplier<MovementType> NATURAL = Sponge.getRegistry().getCatalogRegistry().provideSupplier(MovementType.class, "NATURAL");
 
-    public static final Supplier<TeleportType> PORTAL = Sponge.getRegistry().getCatalogRegistry().provideSupplier(TeleportType.class, "PORTAL");
+    public static final Supplier<MovementType> PLUGIN = Sponge.getRegistry().getCatalogRegistry().provideSupplier(MovementType.class, "PLUGIN");
 
-    public static final Supplier<TeleportType> UNKNOWN = Sponge.getRegistry().getCatalogRegistry().provideSupplier(TeleportType.class, "UNKNOWN");
```

Comment:
Is there a situation where we won't be able to determine what the movement type was? We might want to keep `UNKNOWN` if that's the case.

Reply:
If it is unknown it should be treated as plugin. We control exactly when this is used so we can set the movement. Unknown, by the merits, is unhelpful.